### PR TITLE
feat: Add option in `block-sequence-hyphen-indicator-newline`

### PR DESCRIPTION
**Pull request by sun-yryr**
This is my first pull request to this repository, so please let me know if I am doing something wrong.
Also, sorry if I used broken English.

## why

JetBrains IDE can fix Yaml file as follows.

```yaml
# Before
- !!str "a"
- 'b'
- &anchor "c"
- *anchor
- !!str [ 1, 2 ]
- { a: b }
- [ 1, 2 ]
- name: a
  value: b
- - a
  - b
```
```yaml
# After
- !!str "a"
- 'b'
- &anchor "c"
- *anchor
- !!str [ 1, 2 ]
- { a: b }
- [ 1, 2 ]
-
  name: a
  value: b
-
  - a
  - b

```

Similar to rule `block-sequence-hyphen-indicator-newline` , but `block-sequence-hyphen-indicator-newline` actually adds a line break after all hyphen.

## what

Add a new option `blockMapping` in `block-sequence-hyphen-indicator-newline`.
Defaults, this option is follows global style. If the option is set, it controls the line breaks in the block mapping accordingly.

### CODE REVIEW

**Comment by ota-meshi** [comment]
Thank you for this PR.

Could you please add to the rule documentation how the new options work?

File: tests/fixtures/rules/block-sequence-hyphen-indicator-newline/invalid/yaml-test-suite-for-block-mapping-only/_config.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "options": [
+    "never",
+    {
+      "nestedHyphen": "always",
```
Comment (ota-meshi) [inline tests/fixtures/rules/block-sequence-hyphen-indicator-newline/invalid/yaml-test-suite-for-block-mapping-only/_config.json:5]: Please add test cases that does not use nestedHyphen.

**Comment by sun-yryr** [comment]
Thank you for waiting long time.
Update document and add new test suites. Please review again.

Maybe resolve #243

**Review by ota-meshi — APPROVED** [review]
Thank you!

**Pull request merged**